grails-app/services/org/pih/warehouse/data/MigrationService.groovy

```diff
@@ -440,65 +447,91 @@ class MigrationService {
+                            processProductInventoryTransactions(transactionList, location, migratedBy, results, true)
+                        }
+                    }
+                }
+                log.info("Processing ${multiProductTransactions.size()} multi-product transactions")
+                processProductInventoryTransactions(multiProductTransactions, location, migratedBy, results)
+
+                // Zero out stock for products that were migrated but had no stock initially (hence hand no baseline created)
+                def cleanupAdjustmentEnabled = grailsApplication.config.openboxes.transactions.inventoryBaseline.migration.cleanupAdjustment
+                List<Product> shouldBeZeroedOut = productsToBaseline - currentInventoryBaseline?.transactionEntries?.collect { it.inventoryItem.product }?.unique()
+                if (cleanupAdjustmentEnabled && shouldBeZeroedOut) {
+                    log.info("Check if there is need for 'zeroing out' adjustments for products that had no stock before migration")
+
+                    Map<String, AvailableItem> availableItems = productAvailabilityService.getAvailableItemsAtDateAsMap(
+                            location,
+                            shouldBeZeroedOut
+                    )
+
+                    createCleanupAdjustment(
+                            location,
+                            availableItems,
+                            "After migration from single Product Inventory transaction to Baseline + Adjustment pair " +
+                                    "this transaction is created to ensure that products that had no stock before, " +
+                                    "are currently zeroed out too"
+                    )
+                }
+            } catch (Exception e) {
+                // Catch and exception during migration and log it but do not break the migration process
+                // It will be re-processed later
+                log.error("Error during migration of product inventory transactions at location ${location.name}", e)
             }
 
             // Refresh PA for all products that were migrated
-            productAvailabilityService.triggerRefreshProductAvailability(location.id, products?.id, true)
+            productAvailabilityService.triggerRefreshProductAvailability(location.id, allProducts?.id, true)
 
             createCSVMigrationReportForResults(location, results)
         }
 
+        String batchMessage = "No batching required, all transactions ${performMigration ? 'were' : 'will be'} processed."
+        if (requiresBatching) {
+            batchMessage = "This location has more transactions than specified allowed batch size: ${batchSize}. "
+            batchMessage += performMigration ? "Migration was run only for ${batchSize} most recent transactions." :
+                    "Migration will be run in batches of ${batchSize} most recent transactions."
```

Comment:
Groovy doesn't allow multiline strings? :thinking:

Reply:
For some reason, I thought it would be clearer to separate the ternary operator to that version, but yeah, it is not needed